source/module_hsolver/diago_pxxxgvx.h

```diff
@@ -0,0 +1,37 @@
+#ifndef HSOLVER_DIAGO_PXXXGVX_H
+#define HSOLVER_DIAGO_PXXXGVX_H
+#include <complex>
+#include "module_base/macros.h"
+
+namespace hsolver
+{
+
+#ifdef __MPI
+/**
+ * @brief generalized eigenvalue problem using pdsygvx/pzhegvx/pdsygvx/pzhegvx
+ * 
+ * @param desc the descriptor of scalapack descriptor
+ * @param ncol the number of columns of the H/S matrix in current processor
+ * @param nrow the number of rows of the H/S matrix in current processor
+ * @param nbands the number of bands to be solved
+ * @param h_mat the Hamiltonian matrix
+ * @param s_mat the overlap matrix
+ * @param ekb the eigenvalues
+ * @param wfc_2d the eigenvectors in 2D block cyclic distribution
+ * 
+ */
+
+template <typename T>
+void pxxxgvx_diag(const int* const desc,
```

Comment:
Does p stand for "parameter"?

Reply:
pxxxgvx is the wrapper of pdsygvx_/pzhegvx_/pdsygvx_/pzhegvx_ in scalapack

Reply:
> pxxxgvx is the wrapper of pdsygvx_/pzhegvx_/pdsygvx_/pzhegvx_ in scalapack

I get it. It is "Parallel"